pip_audit/_dependency_source/requirement.py

```diff
@@ -205,9 +205,12 @@ def _fix_file(self, filename: Path, fix_version: ResolvedFixVersion) -> None:
                         r for r in reqs if isinstance(r, InstallRequirement)
                     ]
                     origin_reqs: set[Requirement] = set()
-                    for req, dep in self._collect_cached_deps(filename, list(installed_reqs)):
+                    for dep in self._collect_cached_deps(filename, list(installed_reqs)):
+                        # NOTE(alex): Now we can't map requirements to dependencies, this is broken.
```

Comment:
Yeah, this might be tough. I'll take a look through the `resolvelib` APIs as well, but stashing it in the candidates would be a reasonable approach IMO (since `resolvelib` is generic over the candidate type for purposes like this).